draft-ietf-dnsop-svcb-httpssvc.md

```diff
@@ -778,52 +779,76 @@ responses to the address queries that were issued in parallel.
+the secure handshake MAY differ from the ALPN specified 
+in the SVCB record.
+
+ALPNs are identified by their registered "Identification Sequence"
+(alpn-id), which is a sequence of 1-255 octets.
+
+    alpn-id = 1*255(OCTET)
+
+The presentation value of "alpn" is a comma-separated list of one or
+more `alpn-id`s.  Any commas present in the protocol-id are escaped
 by a backslash:
 
    escaped-octet = %x00-2b / "\," / %x2d-5b / "\\" / %5d-%FF
    escaped-id = 1*255(escaped-octet)
-   transport-value = escaped-id *("," escaped-id)
+   alpn-value = escaped-id *("," escaped-id)
 
-In the wire format for "transport", each protocol-id is prefixed by its
+In the wire format for "alpn", each alpn-id is prefixed by its
 length as a single octet, and these length-value pairs are concatenated
 to form the SvcParamValue.  These pairs MUST exactly fill the SvcParamValue;
 otherwise, the SvcParamValue is malformed.
 
-For "no-default-transport", the presentation and wire format values MUST be
+For "no-default-alpn", the presentation and wire format values MUST be
 empty.
 
 Each scheme that is mapped to SVCB defines a set or registry of allowed
-protocol IDs, and a "default set" of supported IDs, which SHOULD NOT
-be empty.  To determine the set of transport protocols supported by an
-endpoint (the "transport set"), the client parses the "transport" value's
-protocol IDs, and then adds the default set unless the
-"no-default-transport" SvcParamKey is present.  The presence of a value in
-the transport set indicates that this service endpoint, described by
+ALPNs, and a "default set" of supported ALPNs, which SHOULD NOT
+be empty.  To determine the set of protocol suites supported by an
+endpoint (the "alpn set"), the client parses the "alpn" values'
+ALPN IDs, and then adds the default set unless the
+"no-default-alpn" SvcParamKey is present.  The presence of a value in
+the alpn set indicates that this service endpoint, described by
 SvcDomainName and the other parameters (e.g. "port") offers service with
-that transport.
+the protocol suite associated with the ALPN ID.
 
 Clients SHOULD NOT attempt connection to a service endpoint whose
-transport set does not contain any compatible transport protocols.  To ensure
+alpn set does not contain any compatible protocols suites.  To ensure
 consistency of behavior, clients MAY reject the entire SVCB RRSet and fall
 back to basic connection establishment if all of the RRs indicate
-"no-default-transport", even if connection could have succeeded using a
-non-default transport.
+"no-default-alpn", even if connection could have succeeded using a
+non-default alpn.
 
 For compatibility with clients that require default transports,
 zone operators SHOULD ensure that at least one RR in each RRSet supports the
 default transports.
 
+As long as {{ALPN}} is securely negotiated as part of a cryptographic 
+handshake, clients MUST be willing to use "alpn" protocols negotiated
```

Comment:
I think this paragraph should be replaced by the text we previously had on this point: https://github.com/MikeBishop/dns-alt-svc/commit/9c133e81c8932d4017a9cff040577c09273186c7#diff-6a78b06198e2a4dd4bb7dc07a558a5adR696.  That text essentially states this requirement in the opposite way, which I think is clearer.  It could also be combined with the paragraph at Line 797 to reduce repetition.

Reply:
One caution there: "For example, when using a SvcFieldValue with an "alpn" of "h2", the client MAY also" has a "MAY" in a "For example".  The word "could" is better.

Reply:
Done